### ISSUE TITLE
probe summary to targets fails on WTA object where only multi-probe target is the negative control

### ISSUE THREAD

**Issue by andrewrech** [Nanostring-Biostats/GeomxTools#137]
I'm having some trouble understanding what is happening here. I just pulled your HEAD on master into my workflow and am encountering a new issue vs. https://github.com/Nanostring-Biostats/GeomxTools/commit/bdbb648fb2a09f17252f97c69ae74833a34b30b1 in my tests.

When I apply `aggregateCounts` to my object in which the only multi-probe target is `NegProbe-WTX`, I encounter the following error [here](https://github.com/Nanostring-Biostats/GeomxTools/blob/master/R/NanoStringGeoMxSet-aggregate.R#L46):

```
Error in h(simpleError(msg, call)) :
  error in evaluating the argument 'args' in selecting a method for function 'do.call': invalid first argument
Calls: %>% -> .geomxSetQC -> %>% -> aggregateCounts -> do.call -> esBy -> esBy -> .local -> sapply -> sapply -> lapply -> FUN -> FUN -> esApply -> esApply -> multiassign -> assign -> .handleSimpleError -> h
No traceback available
```

I am thinking perhaps

```r
    multiObject <-
      object[fData(object)[["TargetName"]] %in% multiProbeTargs, ]
```

applied to my input object breaks the application of `esApply`?

However, I note while debugging that applying [this](https://github.com/Nanostring-Biaostats/GeomxTools/blob/master/R/NanoStringGeoMxSet-aggregate.R#L32) before creating `multiObject` appears to resolve the issue.

It isn't clear to me how you've implemented the generic `esApply` here nor what `summarizeNegatives` is doing exactly, so I am a bit lost without digging further.

I cannot share the object publicly, I emailed it just now to @NicoleEO.

Thanks

```
R version 4.0.2 (2020-06-22)
Platform: x86_64-koji-linux-gnu (64-bit)
Running under: Amazon Linux 2

Matrix products: default
BLAS/LAPACK: /usr/lib64/R/lib/libRblas.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
 [1] future_1.23.0               magrittr_2.0.1.9000         data.table_1.14.3
 [4] bit64_4.0.5                 bit_4.0.4                   aod_1.3.1
 [7] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0 GenomicRanges_1.42.0
[10] GenomeInfoDb_1.26.7         IRanges_2.24.1              MatrixGenerics_1.2.1
[13] matrixStats_0.61.0-9001     GeomxTools_1.99.4           NanoStringNCTools_1.1.2
[16] ggplot2_3.3.5               S4Vectors_0.28.1            Biobase_2.50.0
[19] BiocGenerics_0.36.1

loaded via a namespace (and not attached):
 [1] nlme_3.1-148           bitops_1.0-7           httr_1.4.2             filelock_1.0.2
 [5] EnvStats_2.4.0         RColorBrewer_1.1-2     numDeriv_2016.8-1.1    tools_4.0.2
 [9] backports_1.3.0        utf8_1.2.2             R6_2.5.1               vipor_0.4.5
[13] lazyeval_0.2.2         DBI_1.1.1              colorspace_2.0-2       withr_2.4.2
[17] tidyselect_1.1.1       compiler_4.0.2         plotly_4.10.0          DelayedArray_0.16.3
[21] labeling_0.4.2         scales_1.1.1           systemfonts_1.0.3      stringr_1.4.0.9000
[25] digest_0.6.28          minqa_1.2.4            XVector_0.30.0         pkgconfig_2.0.3
[29] htmltools_0.5.2        parallelly_1.28.1      lme4_1.1-27.1          fastmap_1.1.0
[33] htmlwidgets_1.5.4      rlang_0.4.12           ggthemes_4.2.4         readxl_1.3.1
[37] farver_2.1.0           generics_0.1.1         jsonlite_1.7.2         crosstalk_1.2.0
[41] car_3.0-12             dplyr_1.0.7            RCurl_1.98-1.5         GenomeInfoDbData_1.2.4
[45] Matrix_1.3-4           Rcpp_1.0.7             ggbeeswarm_0.6.0       munsell_0.5.0
[49] fansi_0.5.0            abind_1.4-5            lifecycle_1.0.1        yaml_2.2.1
[53] stringi_1.7.5          carData_3.0-4          MASS_7.3-51.6          zlibbioc_1.36.0
[57] plyr_1.8.6             grid_4.0.2             listenv_0.8.0          crayon_1.4.2
[61] lattice_0.20-41        cowplot_1.1.1          Biostrings_2.58.0      splines_4.0.2
[65] pillar_1.6.4           ggpubr_0.4.0           uuid_1.0-3             boot_1.3-25
[69] rjson_0.2.20           ggsignif_0.6.3         reshape2_1.4.4         codetools_0.2-16
[73] glue_1.5.0             ggiraph_0.7.10         outliers_0.14          vctrs_0.3.8
[77] nloptr_1.2.2.3         cellranger_1.1.0       gtable_0.3.0           purrr_0.3.4
[81] tidyr_1.1.4            assertthat_0.2.1       broom_0.7.10           rstatix_0.7.0
[85] viridisLite_0.4.0      tibble_3.1.6           pheatmap_1.0.12        lmerTest_3.1-3
[89] beeswarm_0.4.0         globals_0.14.0         ellipsis_0.3.2```

**Comment by andrewrech** [comment]
Diff vs. my working fork if helpful
![Screen Shot 2021-11-12 at 09 03 18](https://user-images.githubusercontent.com/10691207/141505994-400d515e-0e57-4d25-b24a-b4b3f56293dd.png)
l

**Comment by NicoleEO** [comment]
@andrewrech thanks for the data. I have pinpointed the issue to a discrepancy in your GeoMxSet object.

Run `object <- NanoStringGeoMxSet(object, check=FALSE)` to overwrite your old version to a working GeoMxSet object

I did notice that this object was made using R4.0 which may cause an issue with out-of-date dependencies. Below objectN is the new one I made with the above code. objectN runs `aggregateCounts` with the code as-is. I'll also send you back my objectN via email.
![image](https://user-images.githubusercontent.com/20404635/141597877-67a5285b-2c7c-40cb-a5ca-278d2f3034f0.png)
